src/app/components/home-page/NeedHelp.tsx

```diff
@@ -65,14 +77,88 @@ const fallback: Payload['entries'][0] = {
 // eslint-disable-next-line arrow-body-style
 const NeedHelp = () => {
   const { data, loading } = useDataApi<Payload>(
-    'https://www.ebi.ac.uk/ebisearch/ws/rest/ebiweb_training_events?query=timeframe:upcoming AND resources:UniProt The Universal Protein Resource 5544&facets=status:Open&format=json&fieldurl=true&viewurl=true&fields=title,subtitle,description,location,date_time_clean,status&size=1'
+    'https://www.ebi.ac.uk/ebisearch/ws/rest/ebiweb_training_events?query=timeframe:upcoming AND resources:UniProt The Universal Protein Resource 5544&facets=status:Open&format=json&fieldurl=true&viewurl=true&fields=title,subtitle,description,location,city,country,venue,date_time_clean,status&size=1'
   );
 
   const seminar = data?.entries[0] || fallback;
+  const title = `${seminar.fields.title}${
+    seminar.fields.subtitle && `: ${seminar.fields.subtitle}`
+  }`;
+  const url =
+    seminar?.fieldURLs.find(({ name }) => name === 'main')?.value || '';
+  const venue = seminar?.fields.venue[0];
+  const location = seminar?.fields.location[0];
+
+  useStructuredData<Course | Event>(
+    useMemo(() => {
+      const organiser: Consortium = {
+        // TODO: reference to the consortium markup from the footer
+        '@type': 'Consortium',
+        name: 'UniProt consortium',
+        url: 'https://www.uniprot.org',
+      };
+
+      let event: Event | undefined;
+      if (seminar.source === 'ebiweb_training_events') {
+        event = {
+          '@type': 'Event',
+          '@id': `training-event-${seminar.id}`,
+          name: title,
+          startDate:
+            parseDate(seminar.fields.date_time_clean[0])
```

Comment:
mmmh, I guess that it knows "May 6th", and it's assuming midnight UTC. But then when Google server run the code, this parses to the day before, depending on which time zone their server runs on.
Wondering how to tackle that though. I could assume the time is to be displayed in GMT, but then we might have an issue with BST, and when we will resume in-person seminar and those won't happen in the UK, or when PIR or SIB run theirs...
TIme is hard to work with...
I'll give it a think for now. Let me know if you think of another approach

Reply:
OK, so since the parsing of the date is done in the local time of whoever's parsing it (browser, or crawler), I generated the time string without using toISOString but using the methods to get the local time. That way even someone in Japan, or someone in California sees it, it will always render "May 6th".